The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The pre-commit check is a mandatory check, please fix detected issues.

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>ansible/roles/test/files/ptftests/py3/hash_test.py:405:9: F841 local variable 'class_name' is assigned to but never used<br/>ansible/roles/test/files/ptftests/py3/hash_test.py:469:9: F841 local variable 'class_name' is assigned to but never used<br/><br/>flake8...............................................(no files to check)Skipped<br/>flake8 (tests/common2)...............................(no files to check)Skipped<br/>check conditional mark sort..........................(no files to check)Skipped<br/>isort (python).......................................(no files to check)Skipped<br/>black................................................(no files to check)Skipped<br/>mypy.................................................(no files to check)Skipped<br/>pylint...............................................(no files to check)Skipped<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

--- a/ansible/roles/test/files/ptftests/py3/hash_test.py
+++ b/ansible/roles/test/files/ptftests/py3/hash_test.py
@@ -402,7 +402,6 @@ class HashTest(BaseTest):
         '''
         @summary: Check IPv4 route works.
         '''
-        class_name = self.__class__.__name__
         ip_src = self.src_ip_interval.get_random_ip(
         ) if hash_key == 'src-ip' else self.src_ip_interval.get_first_ip()
         ip_dst = self.dst_ip_interval.get_random_ip(
@@ -451,7 +450,7 @@ class HashTest(BaseTest):
             ip_dst=ip_dst,
             ip_proto=ip_proto
         )
-        if class_name == 'HashTest':
+        if isinstance(self, HashTest):
             rcvd_port, rcvd_pkt = retry_call(
                 self.send_and_verify_packets,
                 fargs=[src_port, pkt, masked_exp_pkt, dst_port_lists, logs],
@@ -466,7 +465,6 @@ class HashTest(BaseTest):
         '''
         @summary: Check IPv6 route works.
         '''
-        class_name = self.__class__.__name__
         ip_src = self.src_ip_interval.get_random_ip(
         ) if hash_key == 'src-ip' else self.src_ip_interval.get_first_ip()
         ip_dst = self.dst_ip_interval.get_random_ip(
@@ -517,7 +515,7 @@ class HashTest(BaseTest):
             ip_proto=ip_proto,
             version='IPv6'
         )
-        if class_name == 'HashTest':
+        if isinstance(self, HashTest):
             rcvd_port, rcvd_pkt = retry_call(
                 self.send_and_verify_packets,
                 fargs=[src_port, pkt, masked_exp_pkt, dst_port_lists, logs],

